From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tom Levy <tomlevy93@gmail.com>
Date: Sat, 4 Sep 2021 18:39:56 +1200
Subject: Add configuration option 'show_team_members_publicly'

Default to true, which matches the existing behaviour.

This configuration option can be set to false to only show the "Team
members" to admins.
---
 etc/db-config.yaml                                  | 5 +++++
 webapp/src/Controller/PublicController.php          | 2 +-
 webapp/src/Controller/Team/ScoreboardController.php | 2 +-
 webapp/src/Form/Type/UserRegistrationType.php       | 2 ++
 4 files changed, 9 insertions(+), 2 deletions(-)

diff --git a/etc/db-config.yaml b/etc/db-config.yaml
index 722b91165..8e80cc754 100644
--- a/etc/db-config.yaml
+++ b/etc/db-config.yaml
@@ -177,6 +177,11 @@
             default_value: true
             public: true
             description: Show team members in the interfaces?
+        -   name: show_team_members_publicly
+            type: bool
+            default_value: true
+            public: true
+            description: Show team members on the scoreboard?
         -   name: show_flags
             type: bool
             default_value: true
diff --git a/webapp/src/Controller/PublicController.php b/webapp/src/Controller/PublicController.php
index 5e3c50089..1aafa7f27 100644
--- a/webapp/src/Controller/PublicController.php
+++ b/webapp/src/Controller/PublicController.php
@@ -158,7 +158,7 @@ class PublicController extends BaseController
     public function teamAction(Request $request, int $teamId)
     {
         $team             = $this->em->getRepository(Team::class)->find($teamId);
-        $showTeamMembers  = (bool)$this->config->get('show_team_members');
+        $showTeamMembers  = (bool)$this->config->get('show_team_members') && (bool)$this->config->get('show_team_members_publicly');
         $showFlags        = (bool)$this->config->get('show_flags');
         $showAffiliations = (bool)$this->config->get('show_affiliations');
         $data             = [
diff --git a/webapp/src/Controller/Team/ScoreboardController.php b/webapp/src/Controller/Team/ScoreboardController.php
index 42e64ffcb..9fc854320 100644
--- a/webapp/src/Controller/Team/ScoreboardController.php
+++ b/webapp/src/Controller/Team/ScoreboardController.php
@@ -99,7 +99,7 @@ class ScoreboardController extends BaseController
     public function teamAction(Request $request, int $teamId)
     {
         $team             = $this->em->getRepository(Team::class)->find($teamId);
-        $showTeamMembers  = (bool)$this->config->get('show_team_members');
+        $showTeamMembers  = (bool)$this->config->get('show_team_members') && (bool)$this->config->get('show_team_members_publicly');
         $showFlags        = (bool)$this->config->get('show_flags');
         $showAffiliations = (bool)$this->config->get('show_affiliations');
         $data             = [
diff --git a/webapp/src/Form/Type/UserRegistrationType.php b/webapp/src/Form/Type/UserRegistrationType.php
index 44bab8f17..709a6e44a 100644
--- a/webapp/src/Form/Type/UserRegistrationType.php
+++ b/webapp/src/Form/Type/UserRegistrationType.php
@@ -179,12 +179,14 @@ class UserRegistrationType extends AbstractType
         }
 
         if ($this->config->get('show_team_members')) {
+            $showTeamMembersPublicly = (bool)$this->config->get('show_team_members_publicly');
             $builder
                 ->add('members', TextareaType::class, [
                     'label' => false,
                     'mapped' => false,
                     'attr' => [
                         'placeholder' => 'Team members (optional)',
+                        'title' => $showTeamMembersPublicly ? 'Displayed on the scoreboard.' : 'Only visible to administrators.',
                     ],
                     'required' => false,
                     'mapped' => false,
-- 
2.20.1

